erpnext/manufacturing/doctype/production_plan/production_plan.py

```diff
@@ -300,14 +300,13 @@ def get_production_items(self):
 				item_details.update({
 					"qty": d.planned_qty
 				})
-				item_dict[(d.item_code, d.material_request_item, d.warehouse)] = item_details
+				item_dict[(d.item_code, d.material_request_item, d.warehouse, d.sales_order_item)] = item_details
 			else:
 				item_details.update({
-					"qty": flt(item_dict.get((d.item_code, d.sales_order, d.warehouse),{})
+					"qty": flt(item_dict.get((d.item_code, d.sales_order, d.warehouse, d.sales_order_item),{})
 						.get("qty")) + (flt(d.planned_qty) - flt(d.ordered_qty))
 				})
-				item_dict[(d.item_code, d.sales_order, d.warehouse)] = item_details
-
+				item_dict[(d.item_code, d.sales_order, d.warehouse, d.sales_order_item)] = item_details
```

Comment:
- Do not make any changes in here.
- Override the `make_work_order` frappe call in production_plan.js and call a custom function with your code in the jh app